### DISCUSSION TITLE
UAC权限问题

### DISCUSSION THREAD

**Discussion by fengqing-maker** [jovibor/HexCtrl#83] [Q&A]
现象:在配置属性->连接器->清单文件->UAC执行级别->修改为requireAdministrator;然后编译生成exe,这时候打开exe是不会提示使用管理员权限。
修改方案:在HexCtrl.h里面注释掉#pragma comment(linker,"/manifestdependency:....),然后重新生成exe,这时候打开exe是会提示使用管理员权限。
问题:1.#pragma comment(linker,"/manifestdependency:...)这行代码是干嘛的?(猜测是控件的style之类的)
        2.请问下能否去掉？去掉后有什么影响
        3.您当时为什么要加载这一行代码?

**Comment by jovibor** [comment]
Please use English if possible.
If your question was about this line of code: `#pragma comment(linker,"/manifestdependency:\"type='win32' name='Microsoft.Windows.Common-Controls' version='6.0.0.0' processorArchitecture='*' publicKeyToken='6595b64144ccf1df' language='*'\"")`, it's a standard Microsoft manifest for modern visual styles.
See [here](https://learn.microsoft.com/en-us/windows/win32/controls/cookbook-overview#using-manifests-or-directives-to-ensure-that-visual-styles-can-be-applied-to-applications) for details.

**Reply by fengqing-maker** [reply]
Thank you for your answer.
What i mean is that it doesn't work when i build project with requireAdministrator.
you can see the picture.
![image](https://user-images.githubusercontent.com/55424377/205533567-69deb2a2-ada8-464a-9ea7-b900ce66c3d7.png)

![image](https://user-images.githubusercontent.com/55424377/205534559-dd057525-5be6-4331-b70d-ef12b7b09577.png)

**Reply by jovibor** [reply]
Hmm, frankly saying I don't have an answer to this.
I've tried to set `requireAdministrator` and it worked fine on my machine. But I have UAC totally disabled, because it's causing more inconvenience than help.
I think it might be a more general Windows related question, and how it correlates with compiler/linker options, than the **HexCtrl** question.

Basically, if you ask Visual Studio to generate default MFC Dialog based project, you immediately find these lines in your`stdafx.h`, or `framework.h`:
```cpp
#ifdef _UNICODE
#if defined _M_IX86
#pragma comment(linker,"/manifestdependency:\"type='win32' name='Microsoft.Windows.Common-Controls' version='6.0.0.0' processorArchitecture='x86' publicKeyToken='6595b64144ccf1df' language='*'\"")
#elif defined _M_X64
#pragma comment(linker,"/manifestdependency:\"type='win32' name='Microsoft.Windows.Common-Controls' version='6.0.0.0' processorArchitecture='amd64' publicKeyToken='6595b64144ccf1df' language='*'\"")
#else
#pragma comment(linker,"/manifestdependency:\"type='win32' name='Microsoft.Windows.Common-Controls' version='6.0.0.0' processorArchitecture='*' publicKeyToken='6595b64144ccf1df' language='*'\"")
#endif
#endif
```
So, it's all standard, **HexCtrl** has nothing special here.
